### PR TITLE
GUI: Display aliens, move camera, and more

### PR DESCRIPTION
**Pull request by Sinan-Karakaya**
## **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [X] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR introduces the following:
- Display of the isometric map
- Display of the players with a unique color for each team
- Movement of the camera
- Some various fixes in the reception of the data from the server
- The binary now handles command line arguments
- Instead of using the SFML Socket, it now uses a custom encapsulation of the C Unix sockets
- Prepared the terrain for broadcasting
- Added documentation to header files

## **What is the current behavior?** (You can also link to an open issue here)
The game now launches only if the command line arguments are correct. It will display the isometric map at the current size,
and the different players on the map. The player can move the camera around.

## **What is the new behavior (if this is a feature change)?**
Basic informations are now displayed on screen.

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
The binary needs command line arguments to correctly execute.

### CODE REVIEW

**Review by RoheeAxel — COMMENTED** [review]
Nice we can now see ai

**Review by JasonLAGOUTE — APPROVED** [review]
Nice job, we can see an advancement.

**Review by noerio — APPROVED** [review]
Seems good

File: zappy_gui/src/Utils/RockManager.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const float saturation = 1.0f;
+    const float value = 1.0f;
+
+    const float chroma = value * saturation;
+    const float huePrime = hue / 60.0f;
+    const float x = chroma * (1 - std::abs(std::fmod(huePrime, 2) - 1));
+
+    float red = 0, green = 0, blue = 0;
+
+    if (huePrime >= 0 && huePrime < 1) {
```
Comment (thomasjuin1) [inline zappy_gui/src/Utils/RockManager.cpp:30]: Utilise == au lieu de >= && <

**Review by thomasjuin1 — APPROVED** [review]
Still some thing to change, more docs and it will be fine

File: zappy_gui/src/Utils/RockManager.cpp (hunk 1 above)
Comment (Sinan-Karakaya) [reply zappy_gui/src/Utils/RockManager.cpp:30]: Why?

Comment (JasonLAGOUTE) [reply zappy_gui/src/Utils/RockManager.cpp:30]: Parce que ca fais la même chose et c'est plus logique.

**Review by noerio — APPROVED** [review]
Seems good

**Pull request merged**